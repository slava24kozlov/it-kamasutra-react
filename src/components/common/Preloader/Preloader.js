import React from "react";
import style from './Preloader.module.css';

const Preloader = () => (
  <div className={style.main}/>
)

export default Preloader;