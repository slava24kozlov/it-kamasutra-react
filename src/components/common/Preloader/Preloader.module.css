.main {
    position: absolute;
    left: calc(50vw - 50px);
    transform: translateX(-100%);
    width: 100px;
    height: 100px;
    border: 25px solid #497DDD;
    border-radius: 100px;
    border-right-color: transparent;
    animation: preloader 1s infinite linear;
}

@keyframes preloader {
    from {
        clip-path: circle(76px);
        transform: rotate(0);
    }
    50% {
        clip-path: circle(55px);
    }
    to {
        clip-path: circle(76px);
        transform: rotate(360deg);
    }
}
