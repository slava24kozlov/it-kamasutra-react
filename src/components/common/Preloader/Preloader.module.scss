.main {
    display: flex;
    justify-content: center;
    align-items: center;
    position: fixed;
    top: 0;
    z-index: 5;
    width: 100%;
    height: 100%;
    background-color: rgba(0,0,0,0.5);


    & > div {
        display: inline-block;
        width: 100px;
        height: 100px;
        border: 25px solid #497DDD;
        border-radius: 100px;
        border-right-color: transparent;
        animation: preloader 1s infinite linear;
    }
}

@keyframes preloader {
    from {
        clip-path: circle(76px);
        transform: rotate(0);
    }
    50% {
        clip-path: circle(55px);
    }
    to {
        clip-path: circle(76px);
        transform: rotate(360deg);
    }
}
