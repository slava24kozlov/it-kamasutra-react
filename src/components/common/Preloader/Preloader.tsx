import React from "react";
import style from "./Preloader.module.scss";

const Preloader: React.FC = () => <div aria-label="page preloader" className={style.main}>
    <div/>
</div>;

export default Preloader;
