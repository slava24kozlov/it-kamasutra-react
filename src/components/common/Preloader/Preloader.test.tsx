import { render } from "@testing-library/react";
import React from "react";
import Preloader from "./Preloader";

describe("tests for the Preloader component", () => {
    test("snapshot testing", () => {
        const {container} = render(<Preloader/>);
        expect(container).toMatchSnapshot();
    });
});
