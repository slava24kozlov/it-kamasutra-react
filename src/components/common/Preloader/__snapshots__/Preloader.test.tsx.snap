// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the Preloader component snapshot testing 1`] = `
<div>
  <div
    aria-label="page preloader"
    class="main"
  >
    <div />
  </div>
</div>
`;
